Astrabit-ST/Luminol
# Preferences & Config editors

**Connections**
#43
#45

**Description**
#43 and #45 changed how configuring Luminol worked, but left the configuration editors in an unfinished state. This PR fixes that!
As well as fixing up the config editors, this PR adds support for loading other data formats (but not converting projects, that requires either another out-of-project manager, or changes to the path cache to recognize file extensions when present)

**Testing**
 - (From an old commit where conversion was possible) Load a project, change the format, and reopen it
 - Open the preferences editor and add the path to the `Standard` RTP, then open a project that requires it
 - Edit Game.ini, save, and see the changes
 - Edit Luminol project settings, save, and see changes



**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown -Z build-std=std,panic_abort`
- [x] Run `cargo build --release`
- [x] If applicable, run `trunk build --release`
